src/cloud/docker/docker-extend.md

```diff
@@ -75,6 +75,76 @@ To add a new service version using a `Dockerfile`:
 
 1. Once the build succeeds, test the changes by specifying the [Docker build sources].
 
+## Add a new PHP extension
+
+In addition to built in extensions, you can add PHP extensions to the PHP container.
+
+{:.procedure}
+To add a new PHP extension:
+
+1. Clone the `{{site.data.var.mcd}}` project to your local environment if necessary.
+
+1. On the command line, navigate to the following directory
+   ```bash
+      cd magento-cloud-docker/src/Compose/Php
+   ```
+
+1. Open `ExtensionResolver.php` file, define the required extension inside the `getConfig` method by specifying the extension type and dependency.
+
+   For instance the following block adds the GNUPG extension:
+
+   ```php
+   'gnupg' => [
+     '>=7.0' => [
+     self::EXTENSION_TYPE => self::EXTENSION_TYPE_PECL,
+     self::EXTENSION_OS_DEPENDENCIES => ['libgpgme11-dev'],
+   ],],
+   ```
+   The extension type can be either CORE or PECL, which are defined as `EXTENSION_TYPE_PECL` or `EXTENSION_TYPE_CORE` respectively.
+
+1. Add any required `.ini` files to the PHP FPM container configuration.
+
+   -  On the command line, navigate to FPM image directory.
+
+      ```bash
+      cd magento-cloud-docker/images/php/fpm
```

Comment:
This might error if user is still in the `magento-cloud-docker/src/Compose/Php` directory.